packages/block-library/src/table/edit.js

```diff
@@ -541,6 +544,9 @@ export class TableEdit extends Component {
 			'has-background': !! backgroundColor.color,
 		} );
 
+		// Append the client id to the caption id to ensure no two table block captions can have clashing ids.
+		const captionId = `wp-block-table-caption-${ clientId }`;
```

Comment:
Does that mean that every time a post is saved, a new ID is saved?

Reply:
Fixed by only creating a new id if the existing one is undefined:
https://github.com/WordPress/gutenberg/pull/15554/files#diff-3b49e2fddc9b1f6c3fbd064df1e8af10R513

I did find it quite tricky to try to use an id in this block. I think this is the first time it's been done in a core static block.